lib/internal/http2/core.js

```diff
@@ -351,10 +351,8 @@ function onStreamClose(code) {
     // it completely.
     stream.push(null);
 
-    // Same as net.
-    if (stream.readableLength === 0) {
-      stream.read(0);
-    }
+    if (!stream[kState].didRead && !stream._readableState.resumeScheduled)
+      stream.resume();
```

Comment:
this is slightly different than `read(0)`. Can you please explain why this is needed?
I would prefer if we didn't use `_readableState` here.

Reply:
This is equivalent to what we do in http and is needed because in cases where the user doesn't intend to consume the data (indicated by never having read and not having a pending resume call), we still want to be able to properly destroy the stream.

Reply:
Can you please add a comment about this?

Reply:
Will do later today.